meson.build

```diff
@@ -0,0 +1,132 @@
+
+# i18n module
+i18n = import('i18n')
+
+
+# glib version info
+glib_version = '2.44'
+glib_max_version = glib_version
+glib_min_version = glib_version
+
+# required dependencies
+gtk_dep     = dependency('gtk+-3.0', version: '>= 3.16')
+gio_dep     = dependency('gio-2.0', version: '>= 2.44')
+glib_dep    = dependency('glib-2.0', version: '>= @0@'.format(glib_version))
+json_dep    = dependency('json-glib-1.0', version: '>= 0.8')
+libcurl_dep = dependency('libcurl')
+gthread_dep = dependency('gthread-2.0')
+
+trg_deps = [gtk_dep, glib_dep, gio_dep, json_dep, libcurl_dep, gthread_dep]
+
+# optional dependencies
+rss_dep             = dependency('mrss', version: '>=0.18', required: get_option('mrss'))
+geoip_dep           = dependency('geoip', required: get_option('geoip'))
+libproxy_dep        = dependency('libproxy', required: get_option('libproxy'))
+libnotify_dep       = dependency('libnotify', required: get_option('libnotify'))
+libappindicator_dep = dependency('libappindicator', required: get_option('libappindicator'))
+
+opt_deps = [geoip_dep, libproxy_dep, libnotify_dep, libappindicator_dep]
+foreach opt_dep : opt_deps
+  if opt_dep.found()
+    trg_deps += opt_dep
+  endif
+endforeach
+
+# compiler
+cc = meson.get_compiler('c')
+
+# compiler deps
+trg_deps += cc.find_library('m', required: true)
+
+# nl_langinfo check
+nl_langinfo = cc.has_function('nl_langinfo', prefix: '#include <langinfo.h>')
+if get_option('nl_langinfo').enabled() and not nl_langinfo
+  error('nl_langinfo() specified but not found')
+endif
+nl_langinfo = nl_langinfo and get_option('nl_langinfo').allowed()
+
+
+# config data
+conf_data = configuration_data()
+
+conf_data.set_quoted('PACKAGE_NAME', project_name)
+conf_data.set_quoted('PACKAGE_VERSION', project_version)
+conf_data.set_quoted('GETTEXT_PACKAGE', project_name)
+conf_data.set_quoted('LOCALEDIR', get_option('localedir'))
+
+# debug options
+debug = get_option('debug')
+conf_data.set('DEBUG', debug)
+conf_data.set('G_ENABLE_DEBUG', debug)
+conf_data.set('G_DISABLE_ASSERT', not debug)
+conf_data.set('G_DISABLE_CHECKS', not debug)
+conf_data.set('G_DISABLE_CAST_CHECKS', not debug)
+
+# glib API configuration
+maj_min = glib_max_version.split('.')
+conf_data.set('GLIB_VERSION_MAX_ALLOWED',
+              'GLIB_VERSION_@0@_@1@'.format(maj_min[0], maj_min[1]))
+maj_min = glib_min_version.split('.')
+conf_data.set('GLIB_VERSION_MIN_REQUIRED',
+              'GLIB_VERSION_@0@_@1@'.format(maj_min[0], maj_min[1]))
+
+# compile time features
+conf_data.set('HAVE_RSS', rss_dep.found())
+# TODO: why??
+conf_data.set10('HAVE_GEOIP', geoip_dep.found())
+conf_data.set('HAVE_LIBPROXY', libproxy_dep.found())
+conf_data.set('HAVE_LIBNOTIFY', libnotify_dep.found())
+conf_data.set('ENABLE_NL_LANGINFO', nl_langinfo)
+conf_data.set('HAVE_LIBAPPINDICATOR', libappindicator_dep.found())
+
+# compiler/linker flags
+flags = ['-funsigned-char',
+         '-fstack-protector-strong',
+         '-fPIE',
+         '-fPIC',
```

Comment:
I don't think you need to set PIE/PIC yourself.

meson has
- `-Db_staticpic` (defaults to true)
- `-Db_pie` (defaults to false, so you may just set this in `project()` using the `default_options` kwarg).